cli/azd/pkg/project/project_config.go

```diff
@@ -78,7 +80,8 @@ type ProjectMetadata struct {
 	// Template is a slug that identifies the template and a version. This attribute should be
 	// in every template that we ship.
 	// ex: todo-python-mongo@version
-	Template string
+	Template            string `yaml:"template"`
+	MinSupportedVersion string `yaml:"minSupportedVersion"`
```

Comment:
I didn't use a prefix here because this is the project file for `azd` so it felt like the `azd` or `cli` prefix would be somewhat implied and I didn't want to bake in a name that was probably going to change at some point.

If we end up adding more constraints, we can evolve this file in a compatible way, I think.

Do you think it might make more sense if this was at "top level" in the file instead of under the `metadata` key?

Reply:
It might make sense to support a full version expression and rename to `requiredVersion` and something similar to terraform where the dev could but `>= 1.1.7, < 2.0.0`